### PR TITLE
Mines recode

### PR DESCRIPTION
**Pull request by DylanFPS**
The aim of this PR is to rewrite Mines to better reflect the style of Ranks. This will make developing with mines a lot easier.

# Mines Changes
- Added autoblock, autosmelt and autopickup (#58)
- Refactored the module entirely, with some rewrites here and there.

# API Changes
- Add `getExpToDrop()` method to `BlockBreakEvent`

# New Commands
- `/autosmelt`
- `/autoblock`
- `/autopickup`

# Known Bugs
- ~~Resetting a mine after redefining it requires a reload for the bounds to update, couldn't figure out how to solve this one~~
- The powertools (autosmelt etc) are NOT ready, just a WIP

### CODE REVIEW

**Comment by DylanFPS** [comment]
Good idea. No need to worry - i'll do it.

File: prison-mines/src/main/java/tech/mcprison/prison/mines/managers/MineManager.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
 /**
  * Represents a collection of mines which can be iterated through in a normal <i>for</i> loop
  */
-public class LegacyMineManager implements List<Mine> {
+public class MineManager {
```
Comment (faizaand) [inline prison-mines/src/main/java/tech/mcprison/prison/mines/managers/MineManager.java]: WOOO No longer implementing lists! Nice 👍

File: prison-mines/src/main/java/tech/mcprison/prison/mines/managers/MineManager.java [hunk 2]
```diff
@@ -47,310 +47,34 @@
     int resetCount = 0;
 
     // NPE
-    private HashMap<UUID, LegacyMineManager> players;
+    private HashMap<UUID, MineManager> players;
```
Comment (faizaand) [inline prison-mines/src/main/java/tech/mcprison/prison/mines/managers/MineManager.java]: Why are you creating a whole mine manager for every player? It's probably inefficient, and managers are only supposed to have a single instance during execution.

File: prison-core/src/main/java/tech/mcprison/prison/internal/events/block/BlockBreakEvent.java [hunk 3]
```diff
@@ -35,11 +35,13 @@
     private Location blockLocation;
     private Player player;
     private boolean canceled = false;
+    private int exp;
 
-    public BlockBreakEvent(BlockType block, Location blockLocation, Player player) {
+    public BlockBreakEvent(BlockType block, Location blockLocation, Player player,int xp) {
```
Comment (faizaand) [inline prison-core/src/main/java/tech/mcprison/prison/internal/events/block/BlockBreakEvent.java]: This is a breaking change! Can you restore the old constructor along with this new one?

**Comment by DylanFPS** [comment]
Of course, will do that tomorrow.

**Comment by faizaand** [comment]
Just make sure it works before merging.

**Comment by DylanFPS** [comment]
Will do, although having trouble finding the source of the bounds not
updating bug. Also I figured I’d postpone autotools as I need an entity
system to get that working.

**Pull request merged**